contractcourt/chain_watcher.go

```diff
@@ -974,24 +977,62 @@ func (c *chainWatcher) handleUnknownRemoteState(
 // return zero. This is possible as our output may have been trimmed due to
 // being dust.
 func (c *chainWatcher) toSelfAmount(tx *wire.MsgTx) btcutil.Amount {
-	var selfAmt btcutil.Amount
-	for _, txOut := range tx.TxOut {
+	// There are two main cases we have to handle here. First, in the coop
+	// close case we will always have saved the delivery address we used
+	// whether it was from the upfront shutdown, from the delivery address
+	// requested at close time, or even an automatically generated one. All
+	// coop-close cases can be identified in the following manner:
+	shutdown, _ := c.cfg.chanState.ShutdownInfo()
+	oDeliveryAddr := fn.MapOption(
+		func(i channeldb.ShutdownInfo) lnwire.DeliveryAddress {
+			return i.DeliveryScript.Val
+		})(shutdown)
+
+	// Here we define a function that can pick out the TxOut that
+	// corresponds with our local delivery script from a ShutdownInfo if we
+	// have a ShutdownInfo for this chainWather's channel.
+	//
+	// isDeliveryOutput :: *TxOut -> bool
+	isDeliveryOutput := func(o *wire.TxOut) bool {
+		return fn.ElimOption(
+			oDeliveryAddr,
+			// If we don't have a delivery addr, then the output
+			// can't match it.
+			func() bool { return false },
+			// Otherwise if the PkScript of the TxOut matches our
+			// delivery script then this is a delivery output.
+			func(a lnwire.DeliveryAddress) bool {
+				return slices.Equal(a, o.PkScript)
+			},
+		)
+	}
+
+	// Here we define a function capable of identifying whether an output
+	// belongs to the LND wallet. We use this as a heuristic in the case
+	// where we might be looking for spendable force closure outputs.
+	//
+	// isWalletOutput :: *TxOut -> bool
```

Comment:
I think it makes sense to either introduce this new way of commenting function into the coding docs so that we don't diverge in our coding style ?

Reply:
I'm not sure what the specific ask is here.

The annotation I have here is for informational purposes only to alleviate the need to squint at the function itself to figure out its "shape". The format I use is stolen from Haskell and subtle variations can be found in Standard ML, OCaml, Haskell, Idris, Agda, F#, etc.

In recent years it has been adapted into other more mainstream languages in documentation notes like you can see [here](https://github.com/fantasyland/fantasy-land/tree/v4.0.1), [here](https://github.com/ramda/ramda/wiki/Type-Signatures#named-types) and [here](https://github.com/sanctuary-js/sanctuary). The reason for this is that it's a fairly space efficient way to communicate the shape of the function without diving into the implementation details.

Are you saying that we need a note in the style guidelines to bless this? Or is it to describe in detail how to interpret a standalone type signature? What would you like to see?

Reply:
> Are you saying that we need a note in the style guidelines to bless this?

Yes this is what I meant, because I find it useful.